pyomo/contrib/edi/blackBoxFunctionModel.py

```diff
@@ -16,15 +16,33 @@
 #  ___________________________________________________________________________
 
 import copy
-import numpy as np
 import pyomo
 import pyomo.environ as pyo
 from pyomo.environ import units as pyomo_units
-import scipy.sparse as sps
-from pyomo.contrib.pynumero.interfaces.external_grey_box import (
-    ExternalGreyBoxModel,
-    ExternalGreyBoxBlock,
-)
+
+try:
+    import numpy as np
+except:
+    raise ImportError(
+        "pyomo.contrib.edi requires numpy to enable black box capability, fix with 'pip install numpy' "
+    )
+
+try:
+    import scipy.sparse as sps
+except:
+    raise ImportError(
+        "pyomo.contrib.edi requires scipy to enable black box capability, fix with 'pip install scipy' "
+    )
+
+try:
+    from pyomo.contrib.pynumero.interfaces.external_grey_box import (
+        ExternalGreyBoxModel,
+        ExternalGreyBoxBlock,
+    )
+except:
+    raise ImportError(
+        "pyomo.contrib.edi requires pyomo.contrib.pynumero to be installed to enable black box capability, this should have installed with base pyomo"
+    )
```

Comment:
We have some deferred import infrastructure that I should have pointed you to. It limits the number of packages that get imported until they are actually used which greatly reduces the time it takes to import `pyomo.environ`. Take a look at this file for an example: https://github.com/Pyomo/pyomo/blob/main/pyomo/contrib/pynumero/algorithms/solvers/cyipopt_solver.py

Reply:
This has been resolved in the most recent commit